typescript/sdk/src/deploy/verify/ContractVerifier.ts

```diff
@@ -195,151 +198,142 @@ export class ContractVerifier {
       { apiUrl, chain, result: responseJson.result },
       'Returning result from explorer.',
     );
+
+    await sleep(timeout);
     return responseJson.result;
   }
 
-  // private async isAlreadyVerified(
-  //   chain: ChainName,
-  //   input: ContractVerificationInput,
-  //   verificationLogger: Logger,
-  // ): Promise<boolean> {
-  //   try {
-  //     const result = await this.submitForm(
-  //       chain,
-  //       ExplorerApiActions.GETSOURCECODE,
-  //       verificationLogger,
-  //       {
-  //         address: input.address,
-  //       },
-  //     );
-  //     return !!result[0]?.SourceCode;
-  //   } catch (error: any) {
-  //     // This specific error message is thrown when a contract is not verified.
-  //     if (!error.message.includes(ExplorerApiErrors.NOT_VERIFIED.toString())) {
-  //       verificationLogger.debug(
-  //         { error },
-  //         'Error checking if contract is already verified',
-  //       );
-  //     }
-  //     return false;
-  //   }
-  // }
-
-  private async verifyProxy(
+  private async verify(
     chain: ChainName,
     input: ContractVerificationInput,
     verificationLogger: Logger,
   ): Promise<void> {
-    verificationLogger.debug(`📝 Verifying proxy at ${input.address}...`);
+    const contractType: string = input.isProxy ? 'proxy' : 'implementation';
 
-    try {
-      const proxyGuid = await this.markProxy(chain, input, verificationLogger);
+    verificationLogger.debug(
+      {
+        name: input.name,
+        address: input.address,
+      },
+      `📝 Verifying ${contractType}...`,
+    );
 
-      verificationLogger.trace({ proxyGuid }, 'Checking proxy status...');
-      await this.submitForm(
+    const data = input.isProxy
+      ? this.getProxyData(input)
+      : this.getImplementationData(chain, input, verificationLogger);
+
+    try {
+      const guid: string = await this.submitForm(
         chain,
-        ExplorerApiActions.CHECK_PROXY_STATUS,
+        input.isProxy
+          ? ExplorerApiActions.VERIFY_PROXY
+          : ExplorerApiActions.VERIFY_IMPLEMENTATION,
         verificationLogger,
+        data,
+      );
+
+      verificationLogger.trace(
         {
-          guid: proxyGuid,
+          guid,
+          name: input.name,
+          address: input.address,
         },
+        `Retrieved guid from verified ${contractType}.`,
+      );
+
+      await this.checkStatus(
```

Comment:
is this needed? does `submitForm()` keep checking recursively when `ExplorerApiErrors.VERIFICATION_PENDING`?

Reply:
in the previous implementation, we check `!guid`. is this no longer needed?

Reply:
> is this needed? does `submitForm()` keep checking recursively when `ExplorerApiErrors.VERIFICATION_PENDING`?

good question– i was mostly retrofitting here so didn't consider removal. would like to punt this investigation if you're alright with it

Reply:
> in the previous implementation, we check `!guid`. is this no longer needed?

yes, that was removed because it was never needed